src/auth/User.cc

```diff
@@ -256,7 +256,7 @@ Auth::User::CredentialsCacheStats(StoreEntry *output)
                           Auth::Type_str[auth_user->auth_type],
                           CredentialState_str[auth_user->credentials()],
                           auth_user->ttl(),
```

Comment:
This storeAppendPrintf() function call secretly assumes that ttl() returns something that can be "printed "using `%d`. I recommend refactoring CredentialsCacheStats() internals to use std::ostream instead, preferably in a separate PR. If that is not done, let's at least add an XXX:

```suggestion
                          auth_user->ttl(), // XXX: Here and below, we assume that Ttl can be printed using %d
```